modules/parent-join/src/test/java/org/elasticsearch/join/mapper/ParentJoinFieldMapperTests.java

```diff
@@ -337,11 +342,11 @@ public void testMultipleJoinFields() throws Exception {
                 b.endObject();
             }));
             // Updating the mapping with another join field also fails
-            MapperParsingException exc = expectThrows(
-                MapperParsingException.class,
+            IllegalArgumentException exc = expectThrows(
+                IllegalArgumentException.class,
                 () -> merge(mapperService, mapping(b -> b.startObject("another_join_field").field("type", "join").endObject()))
             );
-            assertThat(exc.getMessage(), containsString("Field [_parent_join] is defined more than once"));
+            assertThat(exc.getMessage(), containsString("Only one [parent-join] field can be defined per index"));
```

Comment:
Now that we include the field names maybe add a check here to ensure they stay in any future refactoring?